extractor/extractor.go

```diff
@@ -0,0 +1,149 @@
+package extractor
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+)
+
+var ErrNotFound = errors.New("not found")
+
+type Extractor interface {
```

Comment:
I'm not sure this is easier to read than Marshal/Unmarshal but it avoids having to do it. Why not use mapstructure like in `go-bitbucket-v1`? 

If it is decided to go this way, it might be good to refactor the other places where marshal/unmarshal is done.

Reply:
> Why not use mapstructure like in go-bitbucket-v1?

Didn't catch what you mean

Reply:
It looks like the extractor does something similar to https://github.com/mitchellh/mapstructure and this lib is already used by https://github.com/gfleury/go-bitbucket-v1. I was wondering if it would be better to use this instead.

Reply:
Let me check how that could be useful for us.

Reply:
@ttetzlaff I have changed implementation to use `mapstructure` and dropped the Extractor code 👍

Reply:
@pavelmemory @ttetzlaff 
Why not use the native json library:` json.marshal`/`json.unmarshal` like in all other places in the code? 
Why add another dependency?

Reply:
@yahavi This is not how the json.marshal/json.unmarshal should be used. We need to refactor other places and used mapstructure instead. But it needs to be done in separate mr.